### PR TITLE
Remove errors from generated protocol buffers code

### PR DESCRIPTION
**Pull request by bbengfort**
Previously we had added an `errors.go` file to `pkg/trisa/api/v1beta1` that allowed us to define methods on the protocol buffer `Error` struct (such as the `Error()` method so that it would implement the `error` interface. This caused problems with CipherTrace's bazel build process. In this PR, we've moved this code out of the `api` package and into a subpackage `pkg/trisa/api/v1beta1/errors` which can be imported with `apierr "github.com/trisacrypto/trisa/pkg/trisa/api/v1beta1/errors"`. The previous handlers are there but instead of methods they are functional utilities and the `api.Error` protocol buffer struct no longer implements the `error` interface`. 

Code and tests have been updated to adapt to this change.

### CODE REVIEW

**Review by DanielSollis — APPROVED** [review]
Looks good to me, just had a couple questions to help me better understand the PR.

File: pkg/trisa/envelope/errors.go [hunk 1]
```diff
@@ -18,4 +18,6 @@ var (
 	ErrCannotEncrypt            = errors.New("cannot encrypt envelope: no cryptographic handler available")
 	ErrCannotSeal               = errors.New("cannot seal envelope: no public key cryptographic handler available")
 	ErrCannotUnseal             = errors.New("cannot unseal envelope: no private key cryptographic handler available")
+	ErrNoOrderingTimesamp       = errors.New("missing ordering timestamp on secure envelope")
+	ErrInvalidOrderingTimestamp = errors.New("could not parse ordering timestamp on secure envelope as RFC3339 timestamp")
```
Comment (rebeccabilbro) [inline pkg/trisa/envelope/errors.go:22]: Nice additions 🙂

**Review by rebeccabilbro — APPROVED** [review]
Thank you @bbengfort for addressing this so quickly; I do hope that it will help CipherTrace in moving to a more routine process of staying synchronized with TRISA. 

Bazel will still likely be an issue for them due to the complexity of the TRISA IVMS101 structs, but for now I have attempted to manually upgrade their version of TRISA to the tagged version v0.99999.1, and was able to update TRISA and run most of their test suite successfully.

**Comment by rebeccabilbro** [comment]
@bbengfort ok if we hold off merging to give Ciphertrace a little additional time to test things on their end?

File: cmd/trisa/main.go [hunk 2]
```diff
@@ -698,7 +699,7 @@ func transfer(c *cli.Context) (err error) {
 	}
 
 	// Is the envelope sealed?
-	if !req.Sealed && (req.Error == nil || req.Error.IsZero()) {
+	if !req.Sealed && (req.Error == nil || apierr.IsZero(req.Error)) {
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply cmd/trisa/main.go:702]: To implement the `error` interface the `api.Error` struct must have an `Error()` method. However, in Go methods can only be defined in the same package that the struct is. CipherTrace wants only generated code in that package for the import to Bazel; so all additional code, including the definition of the `Error()` interface had to be moved to a sub package and no methods can be defined on `api.Error`.

File: pkg/trisa/api/v1beta1/errors/errors.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	UnparseableIdentity      = api.Error_UNPARSEABLE_IDENTITY
+	PrivateInfoWrongFormat   = api.Error_PRIVATE_INFO_WRONG_FORMAT
+	UnparseableTransaction   = api.Error_UNPARSEABLE_TRANSACTION
+	MissingFields            = api.Error_MISSING_FIELDS
+	IncompleteIdentity       = api.Error_INCOMPLETE_IDENTITY
+	ValidationError          = api.Error_VALIDATION_ERROR
+)
+
+// Sygna BVRC rejected error codes
+const (
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/trisa/api/v1beta1/errors/errors.go:53]: Also, BVRC stands for "Beneficiary VASP Rejected Code".

**Comment by bbengfort** [comment]
> @bbengfort ok if we hold off merging to give Ciphertrace a little additional time to test things on their end?

I don't mind at all - I'll move this story into QA while we wait.

**Pull request closed without merging**